### ISSUE TITLE
Steam 5-10MB/s download speed with a wired gigabit connection.

### ISSUE THREAD

**Issue by VladGdev** [ValveSoftware/steam-for-linux#9715]
#### Your system information

* Steam client version (build number or date): 1687386907
* Distribution (e.g. Ubuntu): RebornOS (arch)
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/11854393/steam-logs.tar.gz)
* GPU: AMD

#### Please describe your issue in as much detail as possible:
I have a problem where the steam client only uses about 5-10 MB/s of the down speed. I have a gigabit wired connection, speed test shows 900 Mb/s down. This issue started early April or May and has been occurring the same way my past 3-4 Arch installations. No other program is showing this behavior.

#### Steps for reproducing this issue:

1. Have arch
2. Try to download on steam with a gigabit connection
3. Steam doesn't use all the available down speed

**Comment by MinikPLayer** [comment]
Same thing for me, tried 4 different PC's and laptops, Wifi / Ethernet, Ubuntu / Arch / Fedora, different download servers. In every combination the same thing happens, with 1gbps connection only about 150-300mb/s download with 1gb/s on Windows.

I have an IPv6 network with IPv4 support (disabling IPv6 manually in OS settings doesn't fix the issue).

Tested PCs:
1) Laptop |  i5-7300HQ / 8GB RAM / GTX 1050 / SATA SSD 512GB - Ubuntu 22.04 / Windows 11
2) Laptop | Asus Zephyrus G14 | R7-6800HS / RX 6700S / 40GB RAM / Samsung 980 Pro 2TB - Arch Linux / Fedora / Windows 11
3) Laptop | Redmibook Pro 15 Ryzen 2023 | R7-7840HS / RX 780M / 16GB RAM / Samsung 980 Pro 2TB - Windows 11 / Arch Linux / Fedora
4) PC | R7-7900X / RTX 4070 / 32GB RAM DDR5 / Samsung 980 Pro 2TB - Arch Linux / Fedora / Windows 11

**Comment by MinikPLayer** [comment]
EDIT:
Tested Windows Steam on wine - downloads are at full speed without any problems.

**Comment by MinikPLayer** [comment]
This problem also occurs on the Steam Deck

**Comment by JshKlsn** [comment]
Same issue. Getting 50Mbps on a 1900Mbps connection. Windows works fine.

- Steam client version (build number or date): 1698777785
- Distribution (e.g. Ubuntu): Linux Mint 21.2 Cinnamon
- Opted into Steam client beta?: No

Adding the following code in a steam_dev.cfg file bumped up my download to ~200-240Mbps, but still FAR from 1900Mbps.

`@nClientDownloadEnableHTTP2PlatformLinux 0`
`@fDownloadRateImprovementToAddAnotherConnection 1.0 `

**Comment by susrn** [comment]
> `@nClientDownloadEnableHTTP2PlatformLinux 0` `@fDownloadRateImprovementToAddAnotherConnection 1.0 `

I can confirm adding those tho options in ~/.steam/steam/steam_dev.cfg fixed the issue for me.

**Comment by CatalinPuscoci** [comment]
Disabling "Game File transfer over Local Network" from the settings gave me an extra 50Mbps (now up to 200) right off the bat.

Adding the two options mentioned above then more than tripled it(peaked at 750), and I think I'm bottlenecked by I/O now.

**Comment by AStellarScene** [comment]
This is a big issue. I have hideous download speeds from Steam on Fedora 39. Even with the configuration hacks in place, it is at best 1/3 the speed of Windows. Windows is easily 20X faster than Fedora 39 with those famous commands in place.

800-900 Mbps on Windows, and 30 Mbps on Linux.

@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1.0 

Valve: It is not the Linux community's job to package the launcher nor should it be their job to troubleshoot when fundamental functionality breaks. Yet here we are.

**Comment by deathxxx123** [comment]
Where is this file steam_dev.cfg guis? I dont have it.
Or maybe someone post this file contain, so i can create it.

**Comment by arazilsongweaver** [comment]
This configuration change turned my 60 MBPS download into a 430 MBPS download. The default client settings really need to be adjusted.

For the record, I created a new file named `$HOME/.local/share/Steam` with the following data:

```
@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1.0
```

**Comment by KarnF91** [comment]
> For the record, I created a new file named `$HOME/.local/share/Steam/steam_dev.cfg` with the following data:
> 
> ```
> @nClientDownloadEnableHTTP2PlatformLinux 0
> @fDownloadRateImprovementToAddAnotherConnection 1.0
> ```

This took my download speed from ~18MBps to 50-60MBps.

**Comment by lukasjab** [comment]
for those still suffering from terrible download speeds, try having this in steam_dev.cfg
```
@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1
@cMaxInitialDownloadSources 25
```

**Comment by t0mj** [comment]
Tried these fixes on Steam(Native) in Arch Linux and I'm still getting between 600 kb/s and 5mb/s on downloads on a 1gb service.

**Comment by pullasuti** [comment]
Same issue here with the latest beta, and for a long time before that. Using Pop 22.04 with wired 1Gbit ethernet. Speedtest is giving me consistent 800-900Mbit/s results, but Steam downloads are at max around a few megabytes (or a few dozen megabits). The above mentioned config seems to make no noticeable difference, neither does disabling local downloads or changing the download server location.

Switched to stable, and the issue is still present. 

Ok, it seems if I add the config lines to ~/.local/share/Steam/steam_dev.cfg, they are ignored, but ~/.steam/steam/steam_dev.cfg makes a small difference - at the moment about 3MB/s vs 6MB/s.

Can be seen in the graph, left vs right side. Config change & client restart in the middle, around 3 minutes (don't ask me why there's a 2x difference between the reported rate vs the Steam numbers :smile: ):
![image](https://github.com/ValveSoftware/steam-for-linux/assets/29338403/d910e5db-4517-410a-a53e-f32b902c08ea)

**Comment by katnova** [comment]
I am also having this issue. I have 10Gbps opt connection to my PC. I cannot get more than ~10 MB/s avg. steam_dev.cfg trick has gotten me to a blistering ~45MB/s briefly a few times.

I'm really confused to why I can't seem to get a solid 50MB/s even. Half the time it acts like it doesn't even want to start the download.

**Comment by zoobporsor** [comment]
I have this issue too.
My max download speed is something like 35mb
On windows 11, I can reach that no problem
On Linux, specifically Arch Linux right now, by default I was only able to get 3-4MB

I made a steam_dev.cfg file and put:
@nClientDownloadEnableHTTP2PlatformLinux 0
@fDownloadRateImprovementToAddAnotherConnection 1.0


This helped, now my download speed goes up to 14.5MB,  still its only half as fast as it was on Windows
What is causing this?

**Comment by Justin-Hoffman** [comment]
Duplicate of #10248

**Comment by Silky322** [comment]
Throwing in my 2c as well to add to the list of people affected. The steam_dev.cfg additions (applied either through the cfg or through steam console) don't seem to help and my download speed hovers around 3-5MB/s despite being on a symmetrical 1Gb/s connection and consistently testing at 900+Mb/s to a server in the same city as the steam download region. It might be an intermittent issue as a few weeks ago ran into it as well and after doing some basic troubleshooting (speed test, SMART check, disk speed benchmark) I was able to get back up to ~40MB/s. However trying similar steps today hasn't worked. 

OS: EndeavourOS Linux 6.11.4-arch1-1 x86_64
Client Version: 1726604483 Stable Branch

**Comment by Silky322** [comment]
Through scouring some other threads I've found a workaround/more info on the cause of the issue from this reply https://github.com/ValveSoftware/steam-for-linux/issues/10248#issuecomment-2143747235.  It seems like steam will pick a source server and basically tunnel vision it where it doesn't get any other download sources and instead only connects to the one slow server. 

In my case this is currently cache2-den-iwst.steamcontent.com. If I go into /etc/hosts and block that address the download will hit 5 hosts and download speed will reach theoretical max. If it's unblocked it will only hit the den cache and I'll be stuck at 3Mb/s. See screenshots for the result of download_sources command in both cases. Not sure why only the linux client seems to do this but hopefully this is useful in pinning down the issue.

Unblocked:
![unblocked](https://github.com/user-attachments/assets/a8db846f-2cec-49d5-b471-6d8df4148783)

Blocking DenCache2:
![DenCache2_blocked](https://github.com/user-attachments/assets/ac1cb5e7-57b7-468e-aa77-eb8615779123)

**Comment by TheMelbine** [comment]
> Through scouring some other threads I've found a workaround/more info on the cause of the issue from this reply [#10248 (comment)](https://github.com/ValveSoftware/steam-for-linux/issues/10248#issuecomment-2143747235). It seems like steam will pick a source server and basically tunnel vision it where it doesn't get any other download sources and instead only connects to the one slow server.
> 
> In my case this is currently cache2-den-iwst.steamcontent.com. If I go into /etc/hosts and block that address the download will hit 5 hosts and download speed will reach theoretical max. If it's unblocked it will only hit the den cache and I'll be stuck at 3Mb/s. See screenshots for the result of download_sources command in both cases. Not sure why only the linux client seems to do this but hopefully this is useful in pinning down the issue.
> 
> Unblocked: ![unblocked](https://private-user-images.githubusercontent.com/29170011/378241285-a8db846f-2cec-49d5-b471-6d8df4148783.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NjY2NzEsIm5iZiI6MTczMDU2NjM3MSwicGF0aCI6Ii8yOTE3MDAxMS8zNzgyNDEyODUtYThkYjg0NmYtMmNlYy00OWQ1LWI0NzEtNmQ4ZGY0MTQ4NzgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAyVDE2NTI1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNzU3ZGZmMzlmYmI2M2E5MjNjYjk3MDRiODA1Y2ViMDI3ZTA4YmY5OTc3ODIzOGVhYmZhOWFkNGE4Mzk5NjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._XfneJgRNxLcuizUiuVhMtwZk4lC4N1BfBTh9ykBpCg)
> 
> Blocking DenCache2: ![DenCache2_blocked](https://private-user-images.githubusercontent.com/29170011/378241308-ac1cb5e7-57b7-468e-aa77-eb8615779123.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NjY2NzEsIm5iZiI6MTczMDU2NjM3MSwicGF0aCI6Ii8yOTE3MDAxMS8zNzgyNDEzMDgtYWMxY2I1ZTctNTdiNy00NjhlLWFhNzctZWI4NjE1Nzc5MTIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAyVDE2NTI1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5N2MyNDg3MjE4MjgyZTFjOTkxMDlmYzkyZGMxMzQyZmUxNDkzMDc4NThjZmMzYmU0NjM2MzUzNjhiNGE0ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.U7WaU48dIbQhn2twHP5vOsMMmF0LubIbBOZPuza8clE)

Thanks for your comment, it saved a lot of time. 

p.s. whoever wants to test this. 
For linux: 
1. steam steam://open/console in terminal.
2. in this window type download_source
3. go to /etc/hosts and type in the desired domain. like 0.0.0.0.0.0 example.com. 

For me this has increased the speed by more than 10 times

**Comment by kisak-valve** [comment]
Closing as the general issue appears to have been resolved along with #10248.

If you're still seeing an issue with a specific CDN node, then please open a new issue report with your findings and logs for a Steam dev to ponder.
